controllers/users.go

```diff
@@ -51,68 +51,67 @@ func (uc *userController) UserRegister(ctx echo.Context) error {
 	new_user := models.User{
 		Username:  user.Username,
 		Email:     user.Email,
-		FisrtName: user.FisrtName,
+		FirstName: user.FirstName,
 		LastName:  user.LastName,
 		RoleID:    3,
 	}
 	new_user.SetPassword(user.Password)
 	uid, err := uc.db.CreateUser(&new_user)
 	if err != nil {
+		uc.LogInfo("Failed to create user: " + err.Error())
 		return ctx.JSON(http.StatusConflict, err)
 	}
+	uc.LogInfo("User created: " + new_user.Username)
 	return ctx.JSON(http.StatusCreated, echo.Map{"message": "user created", "uid": uid})
 }
 
-func (uc *userController) CheckUsername(ctx echo.Context) error {
-	var username string
-	if ctx.Bind(&username) != nil {
-		return ctx.JSON(http.StatusBadRequest, echo.Map{"message": "invalid request"})
-	} else if uc.db.CheckUserExists(username) {
-		return ctx.JSON(http.StatusConflict, echo.Map{"message": "username is already taken"})
-	} else {
-		return ctx.JSON(http.StatusOK, echo.Map{"message": "username available"})
-	}
-}
-
 func (uc *userController) UserLogin(ctx echo.Context) error {
-	// Decode the body of request
 	var user UserLoginRequest
 	if ctx.Bind(&user) != nil {
 		return ctx.JSON(http.StatusBadRequest, echo.Map{"message": "invalid request"})
-
 	}
 
-	// Check if user exists
-	db_user, err := uc.db.GetUserByUsername(user.Username)
+	dbUser, err := uc.db.GetUserByUsername(user.Username)
 	if err != nil {
-		if db_user.ID == 0 {
+		if dbUser.ID == 0 {
+			uc.LogInfo("User not found")
 			return ctx.JSON(http.StatusUnauthorized, echo.Map{"message": "user not found"})
-
 		}
+		uc.LogInfo("Error getting user: " + err.Error())
 		return ctx.JSON(http.StatusInternalServerError, echo.Map{"message": "error getting user"})
-
 	}
 
-	// Check if password is correct
-	if db_user.ComparePasswords(user.Password) != nil {
+	if dbUser.ComparePasswords(user.Password) != nil {
+		uc.LogInfo("Wrong password for user: " + dbUser.Username)
```

Comment:
Don't we log it outside when calling from the parent function too? no need for multiple logs for a single error (our logs in the test environment also have some stack traces).

Reply:
👍